src/components/header/OverflowEdgeIndicator.js

```diff
@@ -2,21 +2,40 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+// @flow
+
 import React, { PureComponent, PropTypes } from 'react';
 import classNames from 'classnames';
 
 import './OverflowEdgeIndicator.css';
 
+type Props = {
+  className: string,
+  children: number,
+};
+
 class OverflowEdgeIndicator extends PureComponent {
-  constructor(props) {
+  props: Props;
+  state: {
+    overflowsOnTop: ?boolean,
```

Comment:
Thanks for the merge! Yes I agree, I really didn't want to make them optional. I wrote it in the comment above but perhaps you missed it?
`classNames` wanted them to be `?boolean`, but I find it strange it didn't accept just `boolean`. How would a proper solution to this look like?